roborock/api.py

```diff
@@ -184,10 +193,11 @@ def __init__(self, endpoint: str, device_info: DeviceData, queue_timeout: int =
         self.cache: dict[CacheableAttribute, AttributeCache] = {
             cacheable_attribute: AttributeCache(attr, self) for cacheable_attribute, attr in get_cache_map().items()
         }
-        self._listeners: list[Callable[[str, CacheableAttribute, RoborockBase], None]] = []
         self.is_available: bool = True
         self.queue_timeout = queue_timeout
         self._status_type: type[Status] = ModelStatus.get(self.device_info.model, S7MaxVStatus)
+        if device_info.device.duid not in self._listeners:
+            self._listeners[device_info.device.duid] = ListenerModel({}, self.cache)
```

Comment:
Although this is global now we could still have a variable at class level. Just for simple access to this ListenerModel

Reply:
good point - changed